plugins/services/src/js/pages/EditFrameworkConfiguration.js

```diff
@@ -40,13 +42,18 @@ class EditFrameworkConfiguration extends mixin(StoreMixin) {
     const fullPackage = CosmosPackagesStore.getServiceDetails();
     const packageDetails = new UniversePackage(fullPackage.package);
 
-    // necessary to have review screen same order of tabs as the form
-    const formData = this.reorderResolvedOptions(
+    const schema = packageDetails.getConfig();
+    const formData = getDefaultFormState(
+      schema,
       fullPackage.resolvedOptions,
-      packageDetails.getConfig()
+      schema.definitions
     );
 
-    this.setState({ packageDetails, formData });
+    const defaultConfigWarning = fullPackage.resolvedOptions
+      ? null
+      : "This service was initially deployed to a previous version of DC/OS that did not store service configuration settings. The default package values were used to populate the configuration in this form. Carefully verify the default settings are correct, prior to deploying the new configuration.";
+
+    this.setState({ packageDetails, formData, defaultConfigWarning });
```

Comment:
🤔  Why do we need to store the entire message in the component state? Wouldn't a simple flag be enough?

Reply:
I did this so the `FrameworkConfigurationForm` could stay more general. The `EditFrameworkConfiguration` handles the details of the cosmos API, so I thought it made sense for this component to set the message.